### ISSUE TITLE
*_comfy.safetensors motion / audio issues compared to non-comfy.safetensors file

### ISSUE THREAD

**Issue by sintspiden** [LongCatFan1/longcat-tuner#34]
I recently trained an LTX-2 character LoRA using this fork of musubi-tuner as I heard it had audio working, and better memory management than ai-toolkit. However I'm getting serious problems with the *_comfy.safetensors file compared to the non comfy safetensors file that gets produced. Below is a comparison of the two, and you can see there are the following two issues:

- comfy ui version has incredibly stiff motion compared to the smooth motion of the non-comfyui version. You can see the mouth attempts to move but the lip-sync is totally broken.
- comfy ui version's audio's framerate seems wrong, like it's rushed or something? The non-comfyui version doesn't have this problem, and what's interesting is it seems to be affected by framerate. The clips below are rendered at 25fps, but when I originally rendered them at 24fps this audio problem with the comfy ui safetensors file was even more pronounced, but it didn't occur on the non-comfyui safetensors file.

The likeness is good in the comfyui safetensors file, and I can hear the likeness of the voice starting to come through. However, why is the motion so weirdly stiff like this? I trained a video-only LoRA before this on a different dataset, and had the exact same problem, so I wonder if there is a bug in musubi-tuner, or if I'm doing something wrong?

**Samples:**
These are just quick 512x384 renders to check motion. They're done in comfy against a workflow I don't have up anywhere, but I know it's working solidly for LoRAs trained in ai-toolkit and for rendering without LoRAs. It's using Q6 GGUF quants of dev, and the distilled LoRA at strength 0.6 into an LTXV scheduler in the first pass, then manual sigmas + ic-detailer-lora at 0.6 strength in the upscale pass. Character LoRA is applied at strength 1 to both passes. These were 25 fps.

- Comfy UI safetensors file sample: https://streamable.com/k0ikec
- Non ComfyUI safetensors file sample: https://streamable.com/1sc3of

The settings I trained on were as follows:

- Trained against commit `2c313e588a78984f5ac84394dba299bcc6a2332f` on `ltx-2` branch.
- Dataset was 48 clips of 5 seconds each, all containing the target speaking with no other characters and or audio interfering (decent variation in the clips - all extracted from a show). Resolution trained at was 768x432. Original framerate of the clips is 24fps.
- Trained for 2250 steps (needs another 1k~2k, but this was a check partway through to make sure it's training correctly)
- AdamW8Bit, 1e4 

`dataset_swat.toml`

    [general]
    caption_extension = ".txt"
    batch_size = 1
    num_repeats = 1
    enable_bucket = true
    bucket_no_upscale = false

    [[datasets]]
    video_directory = "/home/me/source/musubi-tuner/datasets/hondo_v1"
    cache_directory = "/home/me/source/musubi-tuner/datasets/hondo_v1/cache"
    resolution = [768, 432]   # [W, H]
    frame_extraction = "head"
    target_frames = [49] # chunk into blocks of this many frames
    source_fps = 24.0
    target_fps = 24.0

Training script:

    DATASET_CONFIG="dataset_swat.toml"
    LTX2_CHECKPOINT="/home/me/source/musubi-tuner/models/ltx-2-19b-dev.safetensors"
    GEMMA_ROOT="/home/me/.cache/huggingface/hub/models--Lightricks--LTX-2/snapshots/f90a633df707f1522c754181183b08193b4f3d9a"
    OUTPUT_DIR="output"
    OUTPUT_NAME="ltx2_hondo_v1"
    
    MAX_TRAIN_STEPS=5000
    SAVE_EVERY_N_STEPS=250
    LTX_2_MODE=av

    NETWORK_DIM=32
    NETWORK_ALPHA=32
    
    BLOCKS_TO_SWAP=25
    VAE_CHUNK_SIZE=16
    
    . venv/bin/activate
    
    python ltx2_cache_latents.py \
      --dataset_config "$DATASET_CONFIG" \
      --ltx2_checkpoint "$LTX2_CHECKPOINT" \
      --device cuda \
      --vae_dtype bf16 \
      --ltx2_mode "$LTX_2_MODE" \
      --vae_chunk_size "$VAE_CHUNK_SIZE"
    
    python ltx2_cache_text_encoder_outputs.py \
      --dataset_config "$DATASET_CONFIG" \
      --ltx2_checkpoint "$LTX2_CHECKPOINT" \
      --gemma_root "$GEMMA_ROOT" \
      --gemma_load_in_4bit \
      --device cuda \
      --mixed_precision bf16 \
      --ltx2_mode "$LTX_2_MODE" \
      --batch_size 1
    
      accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 ltx2_train_network.py \
      --mixed_precision bf16 \
      --dataset_config "$DATASET_CONFIG" \
      --ltx2_checkpoint "$LTX2_CHECKPOINT" \
      --ltx2_mode "$LTX_2_MODE" \
      --gemma_load_in_4bit \
      --gemma_root "$GEMMA_ROOT" \
      --fp8_base \
      --fp8_scaled \
      --blocks_to_swap "$BLOCKS_TO_SWAP" \
      --use_pinned_memory_for_block_swap \
      --gradient_checkpointing \
      --gradient_checkpointing_cpu_offload \
      --flash_attn \
      --network_module networks.lora_ltx2 \
      --network_dim "$NETWORK_DIM" \
      --network_alpha "$NETWORK_ALPHA" \
      --max_train_steps "$MAX_TRAIN_STEPS" \
      --save_every_n_steps "$SAVE_EVERY_N_STEPS" \
      --optimizer_type AdamW8bit \
      --optimizer_args "weight_decay=0.0001" \
      --learning_rate 1e-4 \
      --output_dir "$OUTPUT_DIR" \
      --output_name "$OUTPUT_NAME"

Maybe my config is just whacky, and you can point out something I'm missing, but I'm at a bit of a loss as to what's going on here. Using these same settings on ai-toolkit I get both the likeness of the character, and the motion is normal. The settings might not be exact same, but what I mean is default settings in ai-toolkit (1e-4 AdamW8Bit, rank 32 LoRA etc, 768 resolution).

I'm on Pop_OS 22.04 running NVIDIA 580.82.07 driver with RTX 5060 Ti and 64GB system RAM.

**Comment by AkaneTendo25** [comment]
How are you testing the non-Comfy LoRA? It can't be used directly in ComfyUI (the keys are incompatible), so are you using some other inference framework?

The config itself looks OK. 
The only thing I can think of right now is that there were several recent ComfyUI updates that changed the behavior of LTX-2 inference with unofficial checkpoints. I've heard that non-official checkpoints (including GGUF) might have been disabled or restricted.

What version of ComfyUI are you using?

**Comment by sintspiden** [comment]
ComfyUI version:

    me@pop-os:~/source/ComfyUI$ git log --oneline | head -1
    239ddd33 fix(api-nodes): add price badge for Rodin Gen-2 node (#12512)

ComfyUI-LTXVideo version:

    me@pop-os:~/source/ComfyUI/custom_nodes/ComfyUI-LTXVideo$ git log --oneline | head -1
    82bd963 Merge pull request #411 from Lightricks/pr-2026-02-11

The non-ComfyUI safetensor sample was inferenced from within ComfyUI, which works, but it just doesn't produce accurate likeness visually or audibly, which I guess is to be expected, since it's not compatible. I don't know how else it's supposed to be inferenced? Maybe if you can tell me how that's done, it could test it? I assume the non-comfyui version works given even in comfy it has nice motion and audio.

I'll attempt to download the official fp8 checkpoint and inference it against that instead of the GGUF and report back.

**Comment by AkaneTendo25** [comment]
> I don't know how else it's supposed to be inferenced?

Oh, I just meant that I'm surprised the non-Comfy LoRA actually affects the output - I assumed it would simply be ignored.

> I'll attempt to download the official fp8 checkpoint and inference it against that instead of the GGUF and report back.

That would be great, and it might help pinpoint the issue.

**Comment by sintspiden** [comment]
Oh, right I see what you mean. Yeah, in that case it's probably just the raw output from LTX-2 without the LoRA applied. I didn't realize the non-comfyui version would behave that particular way, and didn't test it without the LoRA enabled at all, so didn't realize.

So, I tested inferencing the comfy.safetensors LoRA against the dev-fp8 checkpoint and it's more or less the same outcome: https://streamable.com/xbld7n

**Comment by sintspiden** [comment]
Looking at this comment https://github.com/ostris/ai-toolkit/issues/684#issuecomment-3828712565 I wonder if it could be related to flash-attn? They mentioned better results using SDPA. I might give that a try for comparison, but first I'm testing out training on the ai-toolkit for where the audio is supposedly fixed, so it might be a few days.

**Comment by sintspiden** [comment]
So, it looks like the most likely explanation after some experiments was a bug in my dataset. I haven't run enough experiments for it to be definitive, but the same dataset gave me a semi-broken LoRA on ai-toolkit too. It was less broken, but broken still.

I'm not sure exactly, but the data started off life as videos with 5.1 surround sound and 24 fps (since it was from a TV show). I took it and cut it into 5 second clips at 121 frames each. Then when I started training, I hit some cryptic error about audio which Claude helped clue me in was because the audio was 5.1 surround sound, so had 6 channels but only two were expected (or something like that), so I used ffmpeg to convert it to stereo and training agreed to start after that. Then I trained over 49 frames. You saw the result I got in musubi-tuner, but the result I got in ai-toolkit was https://streamable.com/92fxas, which is definitely better, but there's still something off about it and this gen is cherry-picked with most having a talking speed that was unnaturally fast.

I downloaded a fresh dataset of 25fps videos with stereo sound and trained in ai-toolkit and musubi-tuner, and I didn't run into the same issues in either tool. For ai-toolkit I got https://streamable.com/8nh2cw and musubi-tuner I got https://streamable.com/wdvmjz. Both of these look fine to me and like the right quality could be achieved by just dialing in the hyper-parameters, so I think we can close this issue.

Maybe if you can provide some insight into a question for me before I close it though... I'm on an RTX 5060 Ti with 64GB system RAM, and the reason I had to set the above training run of the Elon LoRA to only 2 second clips vs the 5 second clips I used in ai-toolkit, is that in musubi-tuner when I tried 5 second clips it just absolutely punished system RAM and swap usage and I was getting like 35s/it, which would have taken me almost 2 days to train. The musubi-tuner LoRA came out significantly worse in terms of movement and likeness, and my guess is that comes down to using 2 second clips, and if I were able to use 5 second clips I'd maybe get the same level of fidelity of movement/likeness I got on the ai-toolkit LoRA. At present I seem to be able to do higher quality in training in ai-toolkit due to this factor alone, which is kind of a shame since the audio clearly is much cleaner in musubi-tuner, and the skin details seem better too...

So I'm wondering does musubi-tuner have knobs available I can modify to reduce the burden on system RAM / swap here???

**Comment by AkaneTendo25** [comment]
The RAM usage usually _should not_ exceed 64 GB RAM, that's strange. Try to remove `--use_pinned_memory_for_block_swap` and/or `--gradient_checkpointing_cpu_offload` from training command - these are only sources of increased RAM usage that I can think of. (If you will be hitting VRAM limits, increase `blocks_to_swap`). Other params in tour configs seem fine.

**Comment by sintspiden** [comment]
Thank you so much! That fixed the memory usage issues. 

Here's a screenshot of my tmux dashboard I use for monitoring VRAM, RAM, swap, and I/O usage during training. The three important things that provide signal here are `free -h`'s 'available' metric, `smem`'s 'PSS' metric, and USED column of `/dev/zram0` + `/swapfile`. 

This was my baseline for reference:

<img width="2560" height="1290" alt="Image" src="https://github.com/user-attachments/assets/f21fe0d1-c527-4c44-90de-fc709b6fa173" />

And this was the improvement after disabling those two options you mentioned. I left the blocks to swap untouched as it was already optimized to maximize my VRAM during training.

<img width="2560" height="1290" alt="Image" src="https://github.com/user-attachments/assets/7024ec3a-0a7c-4ebb-8098-bdb42ab64fa8" />

System RAM usage is basically half what it was and very little swap being used. Training speed also doubled. Really appreciate you helping me figure that out.

I think we can close this issue now?

**Comment by AkaneTendo25** [comment]
That's great! Glad you resolved it.
Yes, it makes sense to close it. If any regression issues appear, feel free to reopen it.
